src/main/java/org/logstash/beats/Server.java

```diff
@@ -58,15 +56,15 @@ public Server listen() throws InterruptedException {
             beatsInitializer = new BeatsInitializer(isSslEnable(), messageListener, clientInactivityTimeoutSeconds);
 
             ServerBootstrap server = new ServerBootstrap();
-            server.group(bossGroup, workGroup)
+            server.group(workGroup)
```

Comment:
Good you raise that, I am currently testing, if would be better to make the beats have his own executor, I am not sure how the pressure would be applied in this case have to reread the channel/thread case. But since we have a bug report concerning connection I am also double checking the keep alive when LS is completely block due to back pressure.

Reply:
Like I've said I wouldn't be able to merge everything in the same `NioEventloop`, if we do this, it will block any `IdleStateHandler`'s write call (the keep alive ack 0). Once we get a way to get back pressure notification when inserting items into the queue it will make the semantic easier and we can remove the executor for the `BeatsHandler`.

- Retry X
- Or Fail early

We could make this behavior configurable at the input level.